src/validate.ts

```diff
@@ -0,0 +1,70 @@
+import type * as typescript from "typescript";
+import { FunctionlessChecker } from "./checker";
+
+/**
+ * Validates a TypeScript SourceFile containing Functionless primitives does not
+ * unsupported syntax.
+ *
+ * @param ts a reference to the TypeScript Server.
+ * @param checker the Program's {@link FunctionlessChecker}.
+ * @param sf the TypeScript SourceFile.
+ * @returns diagnostic errors for the file.
+ */
+export function validate(
+  ts: typeof typescript,
+  checker: FunctionlessChecker,
+  node: typescript.Node,
+  logger?: {
+    info(message: string): void;
+  }
+): typescript.Diagnostic[] {
+  logger?.info("Beginning validation of Functionless semantics");
+
+  return (function visit(node: typescript.Node): typescript.Diagnostic[] {
+    return [
+      ...(checker.isStepFunction(node) ? validateStepFunctionNode(node) : []),
+      ...(visitEachChild(node, visit) ?? []),
```

Comment:
This will process the children of step function either both visit and `validateStepFunction` right? Did you want to short circuit visit once a closure is found? Also, what is the plan to validate function reference from within a closure.

```ts
const math = (x,y) => x + y;
new StepFunction(..., () => math(1,2));
```